module/Finna/src/Finna/Controller/R2SearchController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * R2 Search Controller
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Samuli Sillanpää <samuli.sillanpaa@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+namespace Finna\Controller;
+
+/**
+ * R2 Search Controller
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Samuli Sillanpää <samuli.sillanpaa@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+class R2SearchController extends SearchController
+{
+    use \Finna\Controller\R2ControllerTrait;
+
+    /**
+     * Search class family to use.
+     *
+     * @var string
+     */
+    protected $searchClassId = 'R2';
+
+    /**
+     * Create a new ViewModel.
+     *
+     * @param array $params Parameters to pass to ViewModel constructor.
+     *
+     * @return ViewModel
+     */
+    protected function createViewModel($params = null)
+    {
+        $view = parent::createViewModel($params);
+        $view->searchClassId = $this->searchClassId;
+        return $view;
+    }
+
+    /**
+     * Results action.
+     *
+     * @return mixed
+     */
+    public function searchAction()
+    {
+        $view = parent::resultsAction();
+        $view->setTemplate('search/results');
```

Comment:
Olisiko parempi niin, että tehdään r2:lle omat templatet, jotka toistaiseksi sisältävät vain `<?=$this->render('search/xxx.phtml'); ?>`? Sitten niitä voidaan tarvittaessa kustomoida muokkaamatta controlleria, ja tätä käytäntöä on noudatettu esim. Search2:ssa ennestään.

Reply:
Jeps, selkeämpi näin.